lm-simulator/lm_simulator/crud.py

```diff
@@ -1,129 +1,155 @@
+        ),
     )
-    licenses = session.execute(stmt).scalars().all()
-    return licenses
 
+    db_license_in_use = LicenseInUse(**license_in_use.model_dump())
 
-def delete_license_in_use(
-    session: Session,
-    lead_host: str,
-    user_name: str,
-    quantity: int,
-    license_name: str,
-):
-    """Delete the license_in_use from the database.
+    with handle_errors(
+        "Can't create License In Use",
+        raise_exc_class=HTTPException,
+        exc_builder=lambda exc_class, msg: exc_class(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Can't create License In Use, check the input data",
+        ),
+    ):
+        session.add(db_license_in_use)
+        await session.flush()
+        await session.refresh(db_license_in_use)
 
-    To be able to delete a license_in_use, the license_in_use must exists in the database, if it doesn't
-    then we raise a LicenseNotFound exception.
-    """
-    licenses = _get_licenses_in_database(session, lead_host, user_name, quantity, license_name)
-    if not licenses:
-        raise LicenseNotFound()
+        return LicenseInUseRow.model_validate(db_license_in_use)
 
-    ids_to_delete = [license.id for license in licenses]
-    for id in ids_to_delete:
-        session.execute(delete(LicenseInUse).where(LicenseInUse.id == id))
 
-    session.commit()
+async def list_licenses_in_use(session: Session) -> List[LicenseInUseRow]:
+    """
+    List all LicensesInUse in the database.
+    """
+    query = await session.execute(select(LicenseInUse))
+    db_licenses_in_use = query.scalars().all()
+    return [LicenseInUseRow.model_validate(license_in_use) for license_in_use in db_licenses_in_use]
 
 
-def delete_license(
+async def remove_license_in_use(
     session: Session,
-    license_name: str,
+    id: int,
 ):
-    """Delete the license from the database.
+    """
+    Remove the LicenseInUse from the database.
 
-    To be able to delete a license, the license must exists in the database, if it doesn't
-    then we raise a LicenseNotFound exception.
+    Raises LicenseNotFound if the LicenseInUse does not exist.
     """
-    license = _get_license_available(session, license_name)
-    if not license:
-        raise LicenseNotFound()
+    query = await session.execute(select(LicenseInUse).where(LicenseInUse.id == id))
+    db_license_in_use = query.scalars().one_or_none()
+
+    enforce_defined(
+        db_license_in_use,
+        raise_exc_class=HTTPException,
+        exc_builder=lambda exc_class, msg: exc_class(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="License In Use not found",
+        ),
+    )
 
-    session.execute(delete(License).where(License.name == license_name))
-    session.commit()
+    with handle_errors(
+        "Can't remove License In Use",
+        raise_exc_class=HTTPException,
+        exc_builder=lambda exc_class, msg: exc_class(
+            status_code=status.HTTP_400_BAD_REQUEST,
```

Comment:
Maybe wrong status code? (see above)

Reply:
This one makes sense to be 500, I updated it.